sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,152 @@
+- If the 3rd member initiates a transaction, only the 4th and 5th members are eligible to provide subsequent signatures.
+- Initiating a transaction by the 4th or 5th member is impossible, as there are insufficient subsequent members to complete the signing process.
+
+To summarize, while such a multisig setup might suffice for smaller teams, as the number of required signers increases, it becomes increasingly difficult to create a transaction. This SIP aims to remove these limitations.
+
+# Specification
+
+This section should be interpreted as a patch to the existing "[Specification](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#specification)" section of SIP-005.
+For anything not mentioned here, the rules from SIP-005 still apply.
+
+## Transactions
+
+### Transaction Encoding
+
+#### Transaction Authorization
+
+Add new hash modes `0x05` and `0x07`. These numbers were chosen in order to used in order to maintain the following relationships:
+ - `is_multisig = hash_mode & 0x1`
+ - `is_p2wsh_p2sh = hash_mode & 0x2`
+ - `is_non_sequential_multisig = hash_mode & 0x4`
+
+| Hash mode | Spending Condition | Mainnet version | Hash algorithm |
+| --------- | ------------------ | --------------- | -------------- |
+| `0x00` | Single-signature | 22 | Bitcoin P2PKH |
+| `0x01` | Multi-signature | 20 | Bitcoin redeem script P2SH |
+| `0x02` | Single-signature | 20 | Bitcoin P2WPK-P2SH |
+| `0x03` | Multi-signature | 20 | Bitcoin P2WSH-P2SH |
+| `0x05` | Non-sequential multi-signature | 20 | Bitcoin redeem script P2SH |
+| `0x07` | Non-sequential multi-signature | 20 | Bitcoin P2WSH-P2SH |
+
+#### Transaction Signing and Verifying
+
+The steps for signing a non-sequential multisig transaction (hash modes `0x05` and `0x07`) shall be as follows:
+
+0. Set the spending condition address, and optionally, its signature count.
+1. Clear the other spending condition fields, using the appropriate algorithm below.
+   If this is a sponsored transaction, and the signer is the origin, then set the sponsor spending condition
+   to the "signing sentinel" value (see below).
+2. Serialize the transaction into a byte sequence, and hash it to form an
+   initial `sighash`.
+3. Calculate the `presign-sighash` over the `sighash` by hashing the
+   `sighash` with the authorization type byte (0x04 or 0x05), the fee (as an 8-byte big-endian value),
+   and the nonce (as an 8-byte big-endian value).
+4. Calculate the ECDSA signature over the `presign-sighash` by treating this
+   hash as the message digest.  Note that the signature must be a `libsecp256k1`
+   recoverable signature. Store the message signature and public key encoding byte as a signature auth field.
+5. Repeat step 4 until the signer threshold is reached.
+
+The steps for verifying a non-sequential multisig transaction (hash modes `0x05` and `0x07`) shall be as follows:
+
+0. Extract the public key(s) and signature(s) from the spending condition.
```

Comment:
I think for SIPS/documents in general, we should start numbered lists with `1` rather than the *technically correct* `0`. 
Cosmetic change - I don't feel strongly if it remains starting at `0` though.

Reply:
How are the public keys and signatures encoded?